test-tests/shared/src/test/scala/zio/test/TestAspectSpec.scala

```diff
@@ -0,0 +1,108 @@
+          val result = if (TestPlatform.isJS) succeeded(spec) else ignored(spec)
+          result.map(assert(_, isTrue))
+        },
+        testM("jvm applies test aspect only on jvm") {
+          for {
+            ref    <- Ref.make(false)
+            spec   = test("test")(assert(true, isTrue)) @@ jvm(after(ref.set(true)))
+            _      <- execute(spec)
+            result <- ref.get
+          } yield assert(if (TestPlatform.isJVM) result else !result, isTrue)
+        },
+        testM("jvmOnly runs tests only on the JVM") {
+          val spec   = test("JVM-only")(assert(TestPlatform.isJVM, isTrue)) @@ jvmOnly
+          val result = if (TestPlatform.isJVM) succeeded(spec) else ignored(spec)
+          result.map(assert(_, isTrue))
+        },
+        testM("failure makes a test pass if the result was a failure") {
+          val spec   = test("test")(assert(throw new java.lang.Exception("boom"), isFalse)) @@ failure
+          val result = succeeded(spec)
+          result.map(assert(_, isTrue))
+        },
+        testM("failure makes a test pass if it died with a specified failure") {
+          val spec = test("test")(assert(throw new NullPointerException(), isFalse)) @@ failure(
+            TestAspectSpecHelper.failsWithException[NullPointerException]
+          )
+          succeeded(spec).map(assert(_, isTrue))
+        },
+        testM("failure does not make a test pass if it failed with an unexpected exception") {
+          val spec = test("test")(
+            assert(throw new NullPointerException(), isFalse)
+          ) @@ failure(TestAspectSpecHelper.failsWithException[IllegalArgumentException])
+          failed(spec).map(assert(_, isTrue))
+        },
+        testM("failure does not make a test pass if the specified failure does not match") {
+          val spec = test("test")(assert(throw new RuntimeException(), isFalse)) @@ failure(
+            isCase[TestFailure[String], Cause[String]]("Runtime", {
+              case TestFailure.Runtime(e) => Some(e); case _ => None
+            }, equalTo(Cause.fail("boom")))
+          )
+          failed(spec).map(assert(_, isTrue))
+        },
+        testM("failure makes tests pass on any assertion failure") {
+          val spec = test("test")(assert(true, equalTo(false))) @@ failure
+          succeeded(spec).map(assert(_, isTrue))
+        },
+        testM("failure makes tests pass on an expected assertion failure") {
+          val spec = test("test")(assert(true, equalTo(false))) @@ failure(
+            isCase[TestFailure[Any], Any]("Assertion", {
+              case TestFailure.Assertion(result) => Some(result); case _ => None
+            }, anything)
+          )
+          succeeded(spec).map(assert(_, isTrue))
+        },
+        testM("timeout makes tests fail after given duration") {
+          val spec = (testM("test") {
+            assertM(ZIO.never *> ZIO.unit, equalTo(()))                    //todo what is this doing?
```

Comment:
We're just creating a test that will never terminate to simulate a situation where a user accidentally wrote a non-terminating test (e.g. a bug created a deadlock condition) to make sure that we can timeout the test after the specified duration.

Reply:
thanks for the explanation.